iOS/Source/AddBookmarkController.swift

```diff
@@ -54,9 +61,10 @@ class AddBookmarkController: UIViewController {
         return view
     }()
 
-    init(fetcher: Fetcher) {
+    init(fetcher: Fetcher, delegate: AddBookmarkControllerDelegate? = nil) {
```

Comment:
Woah, for some reason I never considered doing this, it didn't felt right, but I consider is the right approach...(Sending the delegate in the intializer).

I haven't seen it in any Apple code, also.